python/src/trezorlib/cardano.py

```diff
@@ -20,19 +20,139 @@
 from .tools import expect
 
 PROTOCOL_MAGICS = {"mainnet": 764824073, "testnet": 42}
+NETWORK_IDS = {"mainnet": 1, "testnet": 0}
 
 REQUIRED_FIELDS_TRANSACTION = ("inputs", "outputs")
 REQUIRED_FIELDS_INPUT = ("path", "prev_hash", "prev_index")
 
+INCOMPLETE_OUTPUT_ERROR_MESSAGE = "The output is missing some fields"
+
+
+def create_address_parameters(
+    address_type: messages.CardanoAddressType,
+    spending_key_path_str: str,
+    staking_key_path_str: str = None,
+    staking_key_hash_str: str = None,
+    block_index: int = None,
+    tx_index: int = None,
+    certificate_index: int = None,
+) -> messages.CardanoAddressParametersType:
+    staking_key_path = None
+    staking_key_hash = None
+    certificate_pointer = None
+
+    if not _is_known_address_type(address_type):
+        raise ValueError("Unknown address type")
+
+    if address_type == messages.CardanoAddressType.BASE:
+        if staking_key_path_str:
+            staking_key_path = tools.parse_path(staking_key_path_str)
+        elif staking_key_hash_str:
+            staking_key_hash = bytes.fromhex(staking_key_hash_str)
+        else:
+            raise ValueError(
+                "Base address requires a staking key path or a staking key hash"
+            )
+    elif address_type == messages.CardanoAddressType.POINTER:
+        certificate_pointer = create_certificate_pointer(
+            block_index, tx_index, certificate_index
+        )
+
+    return messages.CardanoAddressParametersType(
+        address_type=address_type,
+        spending_key_path=tools.parse_path(spending_key_path_str),
+        staking_key_path=staking_key_path,
+        staking_key_hash=staking_key_hash,
+        certificate_pointer=certificate_pointer,
+    )
+
+
+def _is_known_address_type(address_type: messages.CardanoAddressType) -> bool:
```

Comment:
please make a tuple of all the address types, remove this function, and replace it with `if address_type in ADDRESS_TYPES`

Reply:
Done [here](https://github.com/trezor/trezor-firmware/pull/1112/commits/8c4a5c8a733d9d991405e5ee1c4f9d9bfe9895bf).